commands/chain.go

```diff
@@ -108,3 +121,63 @@ var chainLsCmd = &cmds.Command{
 		}),
 	},
 }
+
+var chainBlockCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Show a filecoin block by its CID",
+		ShortDescription: `Prints the miner, parent weight, height,
+and nonce of a given block. If JSON encoding is specified with the --enc flag,
+all other block properties will be included as well.`,
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("cid", true, false, "CID of block to show"),
+	},
+	Options: []cmdkit.Option{
+		cmdkit.BoolOption("messages", "m", "show messages in block"),
+	},
+	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
+		cid, err := cid.Decode(req.Arguments[0])
+		if err != nil {
+			return err
+		}
+
+		block, err := GetPorcelainAPI(env).ChainGetBlock(req.Context, cid)
+		if err != nil {
+			return err
+		}
+
+		return re.Emit(block)
+	},
+	Type: types.Block{},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, block *types.Block) error {
+			wStr, err := types.FixedStr(uint64(block.ParentWeight))
+			if err != nil {
+				return err
+			}
+
+			_, err = fmt.Fprintf(w, `Block Details
+Miner:  %s
+Weight: %s
+Height: %s
+Nonce:  %s
+Timestamp:  %s
+`,
+				block.Miner,
+				wStr,
+				strconv.FormatUint(uint64(block.Height), 10),
+				strconv.FormatUint(uint64(block.Nonce), 10),
+				strconv.FormatUint(uint64(block.Timestamp), 10),
+			)
+			if err != nil {
+				return err
+			}
+
+			showMessages, _ := req.Options["messages"].(bool)
+			if showMessages == true {
+				_, err = fmt.Fprintf(w, `Messages:  %s`+"\n", block.Messages)
```

Comment:
I'm curious to know what this output would look like. I imagine it would not be very human-friendly, which defeats the purpose of a Text output encoder.

Reply:
in text enc, if show message,  there will be one extra line `Messages:  []`, if not show message, no message line.
in json enc, the code will not run here. no matter show or not show message, it will be the marshall of types.Block, so `messages":null` always exist in json body.
about the messages format, it depend on  fmt.Fprintf in text enc, json Marshall in json enc.
about enc, if you run `go-filecoin --help` it is main option, so if something need update, if it is not only chain command problem.